src/standalone/Soil/rre.jl

```diff
@@ -504,9 +478,25 @@ end
 """
     is_saturated(Y, model::RichardsModel)
 
-A helper function which can be used to indicate whether a layer of soil is 
+A helper function which can be used to indicate whether a layer of soil is
 saturated.
 """
 function is_saturated(Y, model::RichardsModel)
     return @. ClimaLand.heaviside(Y.soil.ϑ_l - model.parameters.ν)
 end
+
+
+"""
+    add_dtopBCdY_to_cache(p::NamedTuple, model::RichardsModel)
+
+Allocate space in `p` for the derivative of the top boundary condition with
+respect to the state variables, and merges it into `p` under the key `dtopBCdY`.
+This is used in the implicit solver for
+`RichardsModel`, where we need to handle the boundary specially in computing
+the Jacobian.
+"""
+function ClimaLand.add_dtopBCdY_to_cache(p::NamedTuple, model::RichardsModel)
```

Comment:
do we always want to add this for RichardsModel, or only if using MoistureStateBC?